tock/tock/templates/uaa_client/login_error.html

```diff
@@ -1,22 +1,22 @@
 {% extends "base.html" %}
+
 {% block content %}
 
-<h2>cloud.gov Login Error</h2>
+  <h2>cloud.gov Login Error</h2>
 
-{% if error_code == "missing_state" or
+  {% if error_code == "missing_state" or
       error_code == "missing_session_state" or
       error_code == "invalid_state" %}
 
-<p>Bad session state</p>
-
-{% elif error_code == "missing_code" %}
+      <p>Bad session state</p>
 
-<p>Missing session code</p>
+  {% elif error_code == "missing_code" %}
 
-{% elif error_code == "authenticate_failed" %}
+    <p>Missing session code</p>
 
-<p>Authentication failed</p>
+  {% else error_code == "authenticate_failed" %}
```

Comment:
Er wait, I'm confused by this... what is this line actually doing?  I understand just an `{% else %}` but I'm unclear on what `{% else error_code == "authenticate_failed" %}` means...

Reply:
Well, it's replacing L15 from before, which used an elif rather than terminating in the `else` as required. I would assume `authenticate_failed` is a response coming back from the fake cloud.gov? @rogeruiz can you provide some additional illumination here?

Reply:
Yes. Those string values are coming from the UAA client but I’m not sure about the `elif` being changed to an `else` here. From the Python docs it looks like it’s `elif`: https://docs.python.org/3/tutorial/controlflow.html

But we could terminate with an empty `else` expression and have a more generic error message.

Reply:
Take a look at `https://docs.djangoproject.com/en/2.0/ref/templates/builtins/`

As I'm reading it, you have to end with the `else` to catch any remainder.  That would jibe with the `Invalid block tag on line 12: 'elif', expected 'endblock'` error I'm seeing, in that it's got know it's done somehow.

I see what you're saying about changing it here though, and agree it might be better to terminate in a generic error, perhaps something like "There was a problem authenticating you: {{ error_code }}" which would also give some indication we may have another case to catch.